lib/chef/resource/aws_cloudsearch_domain.rb

```diff
@@ -0,0 +1,46 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+module AWS
+  class CloudSearch
+    class Domain
+      # The version of the AWS sdk we are using doesn't have a model
+      # object for CloudSearch Domains. This empty class is here to
+      # make the reset of chef-provisioning happy.
+    end
+  end
+end
+
+class Chef::Resource::AwsCloudsearchDomain < Chef::Provisioning::AWSDriver::AWSResource
+  aws_sdk_type ::AWS::CloudSearch::Domain
+  attribute :domain_name, kind_of: String, name_attribute: true
```

Comment:
I am still digging through what this actually does, but adding it breaks this module under test.  If I add it to to the relevant attribute in aws_cache_subnet_group, it also breaks those tests, so my guess is that this only works for things with an actual domain object.